src/operators.jl

```diff
@@ -421,67 +421,70 @@ function _fillwithzeros(arr::AbstractArray{T}) where T
     arr
 end
 
-for op in [:+, :-]; @eval begin
-    function $op(lhs::Number,rhs::AbstractArray{T}) where T<:JuMPTypes
-        ret = similar(rhs, typeof($op(lhs, zero(T))))
-        for I in eachindex(ret)
-            ret[I] = $op(lhs, rhs[I])
-        end
-        ret
-    end
-    function $op(lhs::AbstractArray{T},rhs::Number) where T<:JuMPTypes
-        ret = similar(lhs, typeof($op(zero(T), rhs)))
-        for I in eachindex(ret)
-            ret[I] = $op(lhs[I], rhs)
-        end
-        ret
-    end
-    function $op(lhs::T,rhs::AbstractArray{S}) where {T<:JuMPTypes,S}
-        ret = similar(rhs, typeof($op(lhs, zero(S))))
-        for I in eachindex(ret)
-            ret[I] = $op(lhs, rhs[I])
-        end
-        ret
-    end
-    function $op(lhs::AbstractArray{S},rhs::T) where {T<:JuMPTypes,S}
-        ret = similar(lhs, typeof($op(zero(S), rhs)))
-        for I in eachindex(ret)
-            ret[I] = $op(lhs[I], rhs)
-        end
-        ret
-    end
-end; end
-
-for op in [:*, :/]; @eval begin
-    function $op(lhs::Number,rhs::AbstractArray{T}) where T<:JuMPTypes
-        ret = similar(rhs, typeof($op(lhs, zero(T))))
-        for I in eachindex(ret)
-            ret[I] = $op(lhs, rhs[I])
-        end
-        ret
-    end
-    function $op(lhs::AbstractArray{T},rhs::Number) where T<:JuMPTypes
-        ret = similar(lhs, typeof($op(zero(T), rhs)))
-        for I in eachindex(ret)
-            ret[I] = $op(lhs[I], rhs)
-        end
-        ret
-    end
-    function $op(lhs::T,rhs::AbstractArray{S}) where {T<:JuMPTypes,S}
-        ret = similar(rhs, typeof($op(lhs, zero(S))))
-        for I in eachindex(ret)
-            ret[I] = $op(lhs, rhs[I])
-        end
-        ret
-    end
-    function $op(lhs::AbstractArray{S},rhs::T) where {T<:JuMPTypes,S}
-        ret = similar(lhs, typeof($op(zero(S), rhs)))
-        for I in eachindex(ret)
-            ret[I] = $op(lhs[I], rhs)
-        end
-        ret
-    end
-end; end
+# .+ and .- should be used instead
```

Comment:
How hard is this to address? I don't like the block of commented code, and it's not great either that removing this code didn't break any tests.

Reply:
> How hard is this to address?

I don't mean to say that this code should be replaced by something that use broadcase, I mean to say that this code can be removed as the user shouldn't do `+(scalar, matrix)` but rather `.+(scalar, matrix)`.

> I don't like the block of commented code, and it's not great either that removing this code didn't break any tests.

It broke some tests, that's why I needed to add changes to `parseexpr.jl` but I agree that the coverage of `parseexpr.jl` (currently 37.19%) should be improved before we release v0.19.

Reply:
If this code is no longer needed, then just remove it.